src/DoocsAdapter.cc

```diff
@@ -86,8 +86,8 @@ namespace ChimeraTK {
 
     // connect properties which use the same writable PV to keep the property values consistent
     for(auto& group : doocsAdapter.writeableVariablesWithMultipleProperties) {
-      for(const auto& property : group.second) {
-        auto pc = boost::dynamic_pointer_cast<D_fct>(property);
+      for(const auto& weakProperty : group.second) {
+        auto property = weakProperty.lock();
```

Comment:
You are not checking that the locking worked. Is this always guaranteed?

Reply:
cannot fail, comment added.